paasta_tools/tron_tools.py

```diff
@@ -423,22 +435,51 @@ def format_tron_job_dict(job_config, cluster_fqdn_format, default_paasta_cluster
     return {key: val for key, val in result.items() if val is not None}
 
 
-def load_tron_service_config(service, tron_cluster, load_deployments=True, soa_dir=DEFAULT_SOA_DIR):
-    """Load all configured jobs for a service, and any additional config values."""
-
-    config = service_configuration_lib.read_extra_service_information(service, 'tron-' + tron_cluster, soa_dir)
+def load_tron_instance_config(
+    service: str,
+    instance: str,
+    cluster: str,
+    load_deployments: bool=True,
+    soa_dir: str=DEFAULT_SOA_DIR,
+) -> TronActionConfig:
+    jobs, _ = load_tron_service_config(
+        service=service,
+        tron_cluster=cluster,
+        load_deployments=load_deployments,
+        soa_dir=soa_dir,
+    )
+    requested_job, requested_action = instance.split('.')
+    for job in jobs:
+        if job.get_name() == requested_job:
+            for action in job.get_actions(default_paasta_cluster=cluster):
```

Comment:
My concern about this is that if `cluster` is a Tron cluster, e.g. `prod`, and the action doesn't have a cluster configured, Tron will think that it needs to schedule the task on `paasta-prod`, instead of something like `paasta-norcal-prod` which is correct.

It doesn't matter for local-run, since the task isn't scheduled on Mesos, but it does matter if we want to use load_tron_instance_config for anything that affects Tron configuration.

Reply:
Hmm. You are right, this is a bit concerning and may have some lasting oddities if we go with this approach.

Reply:
I think this is a very valid concern and could have repercussions to our design in the future. Or just plain bugs where `cluster != tron_cluster`, and user-confusion, etc. Let's talk about it IRL and see what we can do short and long term.